### ISSUE TITLE
Android ActiveSync not working and Drafts folder with 190k emails

### ISSUE THREAD

**Issue by ChessSpider** [mailcow/mailcow-dockerized#1310]
Hi,
A colleague of mine only has 74 emails, I confirmed this by going into his  /cur directory of Vmail and count the files there.
however Mailcow says he has 193641 emails and used 800MiB. SOGo says the same thing about space-usage

I have the same problem with my inbox. For both of us Android ActiveSync is not working. Other colleagues of mine do not have this problem. 

I don't know where to start how to debug this issue. Where should I begin?

**Comment by andryyy** [comment]
Can you check `du -hs /var/vmail/domain.tld/user/` for the real size? Also try `docker-compose exec dovecot-mailcow doveadm quota recalc -u user@domain.tld` or even `recalc -A` for all users.

**Comment by andryyy** [comment]
Which Android device + which app? Android is such a mess with its 200 clients and - even worse - 2000 implementations of EAS... it's crazy. You should enable EAS debugging in SOGo and also make sure there are enough workers. Plus: What does "not working" mean exactly? Where does it stop? What error does it print? What's in the SOGo logs when that happens?

**Comment by ChessSpider** [comment]
hmm.. both our drafts folder contains about 190k emails... Do you know if i can just do rm in the Cur directory and recalculate on dovecot for a temporary fix?

**Comment by ChessSpider** [comment]
The last emails in that directory are at 11:59, so it seems to be still syncing somehow. 

With "not working"  I mean that I do a manual refresh, and it just keeps on thinking and at some point just stops. I haven't had the time to seriously debug this issue yet. This has my priority now as my inbox was suddenly full lol.

**Comment by andryyy** [comment]
Ah, are you using GMail? There is an issue with the GMail app and drafts in SOGo.

**Comment by ChessSpider** [comment]
yeah Gmail activesync. Other colleagues of mine don't have that problem though. So I'll just yolo it, delete the drafts folder and recalculate things

**Comment by ChessSpider** [comment]
I did this: 
``docker-compose exec dovecot-mailcow /bin/bash``
``doveadm -v expunge -u username@domain.tld mailbox Drafts  All``


I also had to remove the entire Exchaneg account from Android and re-add it. Problem is now fixed and ActiveSync is working properly on Android again.

**Comment by ChessSpider** [comment]
@andryyy thanks (as always) for your quick response !!

**Comment by ChessSpider** [comment]
@andryyy My phone is doing it again. I just paused ActiveSync for now as my phone was effectively DoS'ing dovecot and causing problems for other users.

In issue #1342 you mentioned you wanted another look if this happened again..  So, I'm all yours, let me know if you want to know or check anything.

Some excerpts from the log. SOGo logentry may not necessarily be about me though; multiple people use ActiveSync 
```
sogo-mailcow_1       | May  7 13:29:31 747f237aef4d sogod [49]: <0x0x56119cbe0eb0[SOGoActiveSyncDispatcher]> Change detected during Sync, we push the content.
dovecot-mailcow_1    | May  7 13:29:30 mail dovecot: imap(email@company.tld): Error: mmap(size=149623144) failed with file /var/vmail/company.tld/email.Drafts/dovecot.index.cache: Cannot allocate memory
```
The mmap error is repeated a lot of times. At that time the Drafts folder was getting quite full already though. After purging I noticed it was growing with about 4-8 mails every second.
